### PR TITLE
[docs] update README.md

### PR DESCRIPTION
**Pull request by rmlmcfadden**
Improve the structure and output formatting of README.md. This should greatly enhance both the readability and clarity of the first thing people see when the visit the repo. The changes include:

- Expanded/reorganized text.
- Formatted code blocks for terminal commands and Python scripts.
- Hyperlinks to Wikipedia articles for various terms/languages/packages.
- New detailed install instructions for Fedora 33.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,101 +1,251 @@
...
+* Multiple interaction potentials can be used in a single simulation for any number of potentials/species.
+  * For example, the He-W interaction can be specified using a Lennard-Jones 12-6 potential, while the W-W interaction can be defined using a Kr-C potential.
+* The scattering integral can be calculated using:
+  * Gauss-Mehler quadrature,
+  * Gauss-Legendre quadrature,
+  * Mendenall-Weller quadrature,
+  * or the MAGIC algorithm.
+* Input files use the [TOML] format, making them both human-readable and easily parsable. 
+* The [Rust] code generates user-friendly error messages, which help pinpoint the cause of errors and provide suggested fixes to the user.
+* The simulation results are formatted as unbiquitous `.csv` files and include:
```
Comment (drobnyjt) [inline README.md:94]: "unbiquitous" - is this a typo?

File: README.md [hunk 2]
```diff
@@ -1,101 +1,251 @@
...
+
+The (optional) [Python] packages can be obtained using:
+
+```bash
+sudo dnf install python3-numpy python3-scipy python3-matplotlib python3-toml python3-shapely
+```
+
+or, alternatively, using `pip3`.
+
+If the [rcpr] is desired, its probably also a good idea to install the following:
```
Comment (drobnyjt) [inline README.md:203]: its -> it's

File: README.md [hunk 3]
```diff
@@ -1,101 +1,251 @@
...
+
+```bash
+sudo dnf install gcc gcc-gfortran cmake lapack lapack-devel blas blas-devel
```
Comment (drobnyjt) [inline README.md:206]: rcpr is at a much, much earlier stage of development, so it's received a lot less attention from me w.r.t. building on multiple systems; does this set of dependencies allow it to compile on a clean installation of Fedora 33? That'd be a great piece of info to have.

**Review by drobnyjt — CHANGES_REQUESTED** [review]
I like all these changes; there's a good deal of markdown functionality here that I was not familiar with! The readability/clarity changes are especially helpful coming from an external contributor. There are a couple potential typos that I caught while proofreading, and I left a question in the comments as well.

File: README.md (hunk 1 above)
Comment (rmlmcfadden) [reply README.md:94]: Whoops - should be "ubiquitous".

File: README.md (hunk 3 above)
Comment (rmlmcfadden) [reply README.md:206]: I can confirm that, with the above, I can build `rustBCA` with `--features cpr_rootfinder_netlib` and  `--features cpr_rootfinder_openblas`, but it's not clear to me that all of the above is necessary - they are just the Fedora equivalents of the packages listed for Ubuntu.

As far as I can tell, with the current build instructions, the full src of LAPACK and OpenBLAS are downloaded and built as part of their Rust bindings. The docs say that when building the (static) system libs can be provided, but this isn't on by default. The extremely long compilation times confirm this. 

On Fedora, static libs are also available and can be installed with:
```bash
sudo dnf install lapack-static openblas-static
```
but `Cargo.toml` would have to be updated to link to them.

**Review by drobnyjt — APPROVED** [review]

**Pull request merged**